timescaledb/how-to-guides/continuous-aggregates/troubleshooting.md

```diff
@@ -12,6 +12,35 @@ with continuous aggregates.
 * Copy this comment at the top of every troubleshooting page
 -->
 
+## Insert irregular data into a continuous aggregate
+Materialized views are generally used with ordered data. If you insert historic
+data, or data that is not related to the current time, you need to refresh
+policies and reevaluate the values that are dragging from past to present.
+
+You can set up an after insert rule for your hypertable or upsert to trigger
+something that can validate what needs to be refreshed as the data is merged.
+
+Let's say you inserted ordered timeframes named A, B, D, and F, and you already
+have a continuous aggregation looking for this data. If you now insert E, you
+need to refresh E and F.  However, if you insert C we'll need to refresh C, D, E
+and F.
+
+For example:
+1.  A, B, D, and F are already materialized in a view with all data.
+1.  To insert C, split the data into `AB` and `DEF` subsets.
+1.  `AB` are consistent and the materialized data is too; you only need to
+    reuse it.
+1.  Insert C, `DEF`, and refresh policies after C.
+
+This can use a lot of resources to process, and if you have any important
+data in the past that also needs to be brought to the present.
```

Comment:
```suggestion
This can use a lot of resources to process, especially if you have any important
data in the past that also needs to be brought to the present.
```